File: LeetCode/Array-String/189. Rotate Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        l = len(nums)
        if l <=1 or k == 0:
            return nums
        if k > l:
            k = k%l
        cc = 0
        for i in range(l - k, l):
            nums.insert(cc, nums[cc+i])
            cc += 1

        del nums[-k:]

File: LeetCode/189. Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k %= n;
        reverseArr(nums, 0, n - k - 1);
        reverseArr(nums, n - k, n - 1);
        reverseArr(nums, 0, n - 1);
    }

    private static void reverseArr(int[] nums, int st, int end) {
        while (st < end) {
            int temp = nums[st];
            nums[st] = nums[end];
            nums[end] = temp;
            st++;
            end--;
        }
    }
}
